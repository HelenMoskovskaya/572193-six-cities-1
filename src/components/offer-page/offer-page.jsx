import React, {PureComponent} from 'react';
import PropTypes from 'prop-types';
import Svg from '../svg/svg.jsx';
import Header from '../header/header.jsx';
import {connect} from 'react-redux';
import {Operation as DataOperation} from '../../reducer/data/data';
import {getLoadStatus, getOfferId, getNeighbourhoodOffers,
  getDetailsOffersForMap, getSortRewiews} from '../../reducer/data/selectors.js';
import {getAuthorizationStatus, getUserData} from '../../reducer/user/selectors.js';
import MapCity from '../map/map.jsx';
import OfferList from '../offer-list/offer-list.jsx';
import PhotoGallery from '../photo-gallery/photo-gallery.jsx';
import DetailsInfo from '../details-info/details-info.jsx';
import HostInfo from '../host-info/host-info.jsx';
import ReviewsList from '../reviews-list/reviews-list.jsx';

class OfferPage extends PureComponent {
  constructor(props) {
    super(props);
  }
  componentDidMount() {
    this.props.loadReviews();
  }

  render() {
    const {offer, isLoadOffers, isAuthorizationRequired,
      userData, neighbourhoodOffers, offers, reviews} = this.props;
    if (isLoadOffers) {
      return <div className="page">
        <Svg />
        <Header isAuthorizationRequired = {isAuthorizationRequired} userData = {userData}/>
        <main className="page__main page__main--property">
          <section className="property">

            <PhotoGallery offer={offer} />

            <div className="property__container container">
              <div className="property__wrapper">

                <DetailsInfo offer={offer} />
                <HostInfo offer={offer} />
                <ReviewsList reviews={reviews}/>

              </div>
            </div>
            <section className="property__map map">

              <MapCity offers={offers} city={offer.city} activeOffer={offer}/>

            </section>
          </section>
          <div className="container">
            <section className="near-places places">
              <h2 className="near-places__title">Other places in the neighbourhood</h2>
              <div className="near-places__list places__list">

                <OfferList offers={neighbourhoodOffers} handleActivateElement={() => null}/>

              </div>
            </section>
          </div>
        </main>
      </div>;
    }
    return null;
  }
}

const mapStateToProps = (state, ownProps) => {
  const id = ownProps.match.params.id;
  return Object.assign({}, ownProps, {
    isLoadOffers: getLoadStatus(state),
    offer: getOfferId(state, id),
    offers: getDetailsOffersForMap(state, id),
    isAuthorizationRequired: getAuthorizationStatus(state),
    userData: getUserData(state),
    neighbourhoodOffers: getNeighbourhoodOffers(state, id),
    reviews: getSortRewiews(state),
  });
};

const mapDispatchToProps = (dispatch, ownProps) => ({
  loadReviews: () => {
    dispatch(DataOperation.loadReviews(ownProps.match.params.id));
  },
});

OfferPage.propTypes = {
  offer: PropTypes.shape({
    city: PropTypes.shape({
      name: PropTypes.string.isRequired,
      location: PropTypes.shape({
        latitude: PropTypes.number.isRequired,
        longitude: PropTypes.number.isRequired,
        zoom: PropTypes.number.isRequired
      })
    }),
    previewImage: PropTypes.string.isRequired,
    images: PropTypes.arrayOf(PropTypes.string.isRequired).isRequired,
    title: PropTypes.string.isRequired,
    isFavorite: PropTypes.bool.isRequired,
    isPremium: PropTypes.bool.isRequired,
    rating: PropTypes.number.isRequired,
    type: PropTypes.string.isRequired,
    bedrooms: PropTypes.number.isRequired,
    maxAdults: PropTypes.number.isRequired,
    price: PropTypes.number.isRequired,
    goods: PropTypes.arrayOf(PropTypes.string.isRequired).isRequired,
    host: PropTypes.shape({
      id: PropTypes.number.isRequired,
      name: PropTypes.string.isRequired,
      isPro: PropTypes.bool.isRequired,
      avatarUrl: PropTypes.string.isRequired
    }),
    description: PropTypes.string.isRequired,
    location: PropTypes.shape({
      latitude: PropTypes.number.isRequired,
      longitude: PropTypes.number.isRequired,
      zoom: PropTypes.number.isRequired
    }),
    id: PropTypes.number.isRequired
  }),
  isLoadOffers: PropTypes.bool.isRequired,
  isAuthorizationRequired: PropTypes.bool.isRequired,
  userData: PropTypes.shape({
    avatarUrl: PropTypes.string,
    email: PropTypes.string,
    id: PropTypes.number,
    isPro: PropTypes.bool,
    name: PropTypes.string
  }),
};

export {OfferPage};
export default connect(mapStateToProps, mapDispatchToProps)(OfferPage);
