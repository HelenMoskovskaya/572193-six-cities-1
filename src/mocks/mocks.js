export const mockOffers = [
  {
    city: {
      name: `Amsterdam`,
      location: {
        latitude: 51.225402,
        longitude: 6.776314,
        zoom: 13
      }
    },
    previewImage: `https://es31-server.appspot.com/six-cities/static/hotel/1.jpg`,
    images: [
      `https://es31-server.appspot.com/six-cities/static/hotel/18.jpg`,
      `https://es31-server.appspot.com/six-cities/static/hotel/4.jpg`,
      `https://es31-server.appspot.com/six-cities/static/hotel/12.jpg`,
      `https://es31-server.appspot.com/six-cities/static/hotel/20.jpg`,
      `https://es31-server.appspot.com/six-cities/static/hotel/17.jpg`,
      `https://es31-server.appspot.com/six-cities/static/hotel/15.jpg`,
      `https://es31-server.appspot.com/six-cities/static/hotel/2.jpg`,
      `https://es31-server.appspot.com/six-cities/static/hotel/8.jpg`,
      `https://es31-server.appspot.com/six-cities/static/hotel/5.jpg`,
      `https://es31-server.appspot.com/six-cities/static/hotel/6.jpg`,
      `https://es31-server.appspot.com/six-cities/static/hotel/1.jpg`,
      `https://es31-server.appspot.com/six-cities/static/hotel/7.jpg`,
      `https://es31-server.appspot.com/six-cities/static/hotel/19.jpg`,
      `https://es31-server.appspot.com/six-cities/static/hotel/14.jpg`
    ],
    title: `Waterfront with extraordinary view`,
    isFavorite: false,
    isPremium: true,
    rating: 2.7,
    type: `hotel`,
    bedrooms: 2,
    maxAdults: 8,
    price: 393,
    goods: [
      `Breakfast`,
      `Washer`,
      `Laptop friendly workspace`
    ],
    host: {
      id: 25,
      name: `Angelina`,
      isPro: true,
      avatarUrl: `img/avatar-angelina.jpg`
    },
    description: `I am happy to welcome you to my apartment in the city center! Three words: location, cosy and chic!`,
    location: {
      latitude: 51.214402,
      longitude: 6.764314000000001,
      zoom: 16
    },
    id: 1
  },
  {
    city: {
      name: `Amsterdam`,
      location: {
        latitude: 51.225402,
        longitude: 6.776314,
        zoom: 13
      }
    },
    previewImage: `https://es31-server.appspot.com/six-cities/static/hotel/1.jpg`,
    images: [
      `https://es31-server.appspot.com/six-cities/static/hotel/18.jpg`,
      `https://es31-server.appspot.com/six-cities/static/hotel/4.jpg`,
      `https://es31-server.appspot.com/six-cities/static/hotel/12.jpg`,
      `https://es31-server.appspot.com/six-cities/static/hotel/20.jpg`,
      `https://es31-server.appspot.com/six-cities/static/hotel/17.jpg`,
      `https://es31-server.appspot.com/six-cities/static/hotel/15.jpg`,
      `https://es31-server.appspot.com/six-cities/static/hotel/2.jpg`,
      `https://es31-server.appspot.com/six-cities/static/hotel/8.jpg`,
      `https://es31-server.appspot.com/six-cities/static/hotel/5.jpg`,
      `https://es31-server.appspot.com/six-cities/static/hotel/6.jpg`,
      `https://es31-server.appspot.com/six-cities/static/hotel/1.jpg`,
      `https://es31-server.appspot.com/six-cities/static/hotel/7.jpg`,
      `https://es31-server.appspot.com/six-cities/static/hotel/19.jpg`,
      `https://es31-server.appspot.com/six-cities/static/hotel/14.jpg`
    ],
    title: `Waterfront with extraordinary view`,
    isFavorite: false,
    isPremium: true,
    rating: 2.7,
    type: `hotel`,
    bedrooms: 2,
    maxAdults: 8,
    price: 393,
    goods: [
      `Breakfast`,
      `Washer`,
      `Laptop friendly workspace`
    ],
    host: {
      id: 25,
      name: `Angelina`,
      isPro: true,
      avatarUrl: `img/avatar-angelina.jpg`
    },
    description: `I am happy to welcome you to my apartment in the city center! Three words: location, cosy and chic!`,
    location: {
      latitude: 51.214402,
      longitude: 6.764314000000001,
      zoom: 16
    },
    id: 3
  }
];

export const mockOffer = {
  city: {
    name: `Amsterdam`,
    location: {
      latitude: 51.225402,
      longitude: 6.776314,
      zoom: 13
    }
  },
  previewImage: `https://es31-server.appspot.com/six-cities/static/hotel/1.jpg`,
  images: [
    `https://es31-server.appspot.com/six-cities/static/hotel/18.jpg`,
    `https://es31-server.appspot.com/six-cities/static/hotel/4.jpg`,
    `https://es31-server.appspot.com/six-cities/static/hotel/12.jpg`,
    `https://es31-server.appspot.com/six-cities/static/hotel/20.jpg`,
    `https://es31-server.appspot.com/six-cities/static/hotel/17.jpg`,
    `https://es31-server.appspot.com/six-cities/static/hotel/15.jpg`,
    `https://es31-server.appspot.com/six-cities/static/hotel/2.jpg`,
    `https://es31-server.appspot.com/six-cities/static/hotel/8.jpg`,
    `https://es31-server.appspot.com/six-cities/static/hotel/5.jpg`,
    `https://es31-server.appspot.com/six-cities/static/hotel/6.jpg`,
    `https://es31-server.appspot.com/six-cities/static/hotel/1.jpg`,
    `https://es31-server.appspot.com/six-cities/static/hotel/7.jpg`,
    `https://es31-server.appspot.com/six-cities/static/hotel/19.jpg`,
    `https://es31-server.appspot.com/six-cities/static/hotel/14.jpg`
  ],
  title: `Waterfront with extraordinary view`,
  isFavorite: false,
  isPremium: true,
  rating: 2.7,
  type: `hotel`,
  bedrooms: 2,
  maxAdults: 8,
  price: 393,
  goods: [
    `Breakfast`,
    `Washer`,
    `Laptop friendly workspace`
  ],
  host: {
    id: 25,
    name: `Angelina`,
    isPro: true,
    avatarUrl: `img/avatar-angelina.jpg`
  },
  description: `I am happy to welcome you to my apartment in the city center! Three words: location, cosy and chic!`,
  location: {
    latitude: 51.214402,
    longitude: 6.764314000000001,
    zoom: 16
  },
  id: 1
};

export const cityMock = {
  name: `Amsterdam`,
  location: {
    latitude: 51.225402,
    longitude: 6.776314,
    zoom: 13
  }
};

export const cities = [`Amsterdam`, `Paris`, `Colone`];

export const userData = {
  id: 1,
  email: `Oliver.conner@gmail.com`,
  name: `Oliver.conner`,
  avatarUrl: `img/1.png`,
  isPro: false
};

export const mockFavorites = {
  "Amsterdam": mockOffers,
  "Colone": mockOffers
};

export const reviewsMock = [
  {
    id: 1,
    user: {
      id: 4,
      isPro: false,
      name: `Max`,
      avatarUrl: `img/1.png`
    },
    rating: 4,
    comment: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.`,
    date: `2019-05-08T14:13:56.569Z`
  }
];

export const reviewItemMock = {
  id: 1,
  user: {
    id: 4,
    isPro: false,
    name: `Max`,
    avatarUrl: `img/1.png`
  },
  rating: 4,
  comment: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.`,
  date: `2019-05-08T14:13:56.569Z`
};
