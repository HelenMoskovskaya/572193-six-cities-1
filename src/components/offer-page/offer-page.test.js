import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import {OfferPage} from './offer-page.jsx';
import {mockOffer, mockOffers, reviewsMock, userData} from '../../mocks/mocks.js';


const renderer = new ShallowRenderer();

describe(`OfferPage`, () => {
  it(`renders correctly`, () => {
    renderer.render(
        <OfferPage
          offer={mockOffer}
          isLoadOffers={true}
          isAuthorizationRequired={false}
          userData={userData}
          neighbourhoodOffers={mockOffers}
          offers={mockOffers}
          reviews={reviewsMock}
        />
    );
    const tree = renderer.getRenderOutput();
    expect(tree).toMatchSnapshot();
  });
});
