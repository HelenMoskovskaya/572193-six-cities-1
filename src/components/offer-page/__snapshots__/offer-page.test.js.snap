// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OfferPage renders correctly 1`] = `
<div
  className="page"
>
  <Svg />
  <Header
    isAuthorizationRequired={false}
    userData={
      Object {
        "avatarUrl": "img/1.png",
        "email": "Oliver.conner@gmail.com",
        "id": 1,
        "isPro": false,
        "name": "Oliver.conner",
      }
    }
  />
  <main
    className="page__main page__main--property"
  >
    <section
      className="property"
    >
      <PhotoGallery
        offer={
          Object {
            "bedrooms": 2,
            "city": Object {
              "location": Object {
                "latitude": 51.225402,
                "longitude": 6.776314,
                "zoom": 13,
              },
              "name": "Amsterdam",
            },
            "description": "I am happy to welcome you to my apartment in the city center! Three words: location, cosy and chic!",
            "goods": Array [
              "Breakfast",
              "Washer",
              "Laptop friendly workspace",
            ],
            "host": Object {
              "avatarUrl": "img/avatar-angelina.jpg",
              "id": 25,
              "isPro": true,
              "name": "Angelina",
            },
            "id": 1,
            "images": Array [
              "https://es31-server.appspot.com/six-cities/static/hotel/18.jpg",
              "https://es31-server.appspot.com/six-cities/static/hotel/4.jpg",
              "https://es31-server.appspot.com/six-cities/static/hotel/12.jpg",
              "https://es31-server.appspot.com/six-cities/static/hotel/20.jpg",
              "https://es31-server.appspot.com/six-cities/static/hotel/17.jpg",
              "https://es31-server.appspot.com/six-cities/static/hotel/15.jpg",
              "https://es31-server.appspot.com/six-cities/static/hotel/2.jpg",
              "https://es31-server.appspot.com/six-cities/static/hotel/8.jpg",
              "https://es31-server.appspot.com/six-cities/static/hotel/5.jpg",
              "https://es31-server.appspot.com/six-cities/static/hotel/6.jpg",
              "https://es31-server.appspot.com/six-cities/static/hotel/1.jpg",
              "https://es31-server.appspot.com/six-cities/static/hotel/7.jpg",
              "https://es31-server.appspot.com/six-cities/static/hotel/19.jpg",
              "https://es31-server.appspot.com/six-cities/static/hotel/14.jpg",
            ],
            "isFavorite": false,
            "isPremium": true,
            "location": Object {
              "latitude": 51.214402,
              "longitude": 6.764314000000001,
              "zoom": 16,
            },
            "maxAdults": 8,
            "previewImage": "https://es31-server.appspot.com/six-cities/static/hotel/1.jpg",
            "price": 393,
            "rating": 2.7,
            "title": "Waterfront with extraordinary view",
            "type": "hotel",
          }
        }
      />
      <div
        className="property__container container"
      >
        <div
          className="property__wrapper"
        >
          <Memo(ConnectFunction)
            offer={
              Object {
                "bedrooms": 2,
                "city": Object {
                  "location": Object {
                    "latitude": 51.225402,
                    "longitude": 6.776314,
                    "zoom": 13,
                  },
                  "name": "Amsterdam",
                },
                "description": "I am happy to welcome you to my apartment in the city center! Three words: location, cosy and chic!",
                "goods": Array [
                  "Breakfast",
                  "Washer",
                  "Laptop friendly workspace",
                ],
                "host": Object {
                  "avatarUrl": "img/avatar-angelina.jpg",
                  "id": 25,
                  "isPro": true,
                  "name": "Angelina",
                },
                "id": 1,
                "images": Array [
                  "https://es31-server.appspot.com/six-cities/static/hotel/18.jpg",
                  "https://es31-server.appspot.com/six-cities/static/hotel/4.jpg",
                  "https://es31-server.appspot.com/six-cities/static/hotel/12.jpg",
                  "https://es31-server.appspot.com/six-cities/static/hotel/20.jpg",
                  "https://es31-server.appspot.com/six-cities/static/hotel/17.jpg",
                  "https://es31-server.appspot.com/six-cities/static/hotel/15.jpg",
                  "https://es31-server.appspot.com/six-cities/static/hotel/2.jpg",
                  "https://es31-server.appspot.com/six-cities/static/hotel/8.jpg",
                  "https://es31-server.appspot.com/six-cities/static/hotel/5.jpg",
                  "https://es31-server.appspot.com/six-cities/static/hotel/6.jpg",
                  "https://es31-server.appspot.com/six-cities/static/hotel/1.jpg",
                  "https://es31-server.appspot.com/six-cities/static/hotel/7.jpg",
                  "https://es31-server.appspot.com/six-cities/static/hotel/19.jpg",
                  "https://es31-server.appspot.com/six-cities/static/hotel/14.jpg",
                ],
                "isFavorite": false,
                "isPremium": true,
                "location": Object {
                  "latitude": 51.214402,
                  "longitude": 6.764314000000001,
                  "zoom": 16,
                },
                "maxAdults": 8,
                "previewImage": "https://es31-server.appspot.com/six-cities/static/hotel/1.jpg",
                "price": 393,
                "rating": 2.7,
                "title": "Waterfront with extraordinary view",
                "type": "hotel",
              }
            }
          />
          <HostInfo
            offer={
              Object {
                "bedrooms": 2,
                "city": Object {
                  "location": Object {
                    "latitude": 51.225402,
                    "longitude": 6.776314,
                    "zoom": 13,
                  },
                  "name": "Amsterdam",
                },
                "description": "I am happy to welcome you to my apartment in the city center! Three words: location, cosy and chic!",
                "goods": Array [
                  "Breakfast",
                  "Washer",
                  "Laptop friendly workspace",
                ],
                "host": Object {
                  "avatarUrl": "img/avatar-angelina.jpg",
                  "id": 25,
                  "isPro": true,
                  "name": "Angelina",
                },
                "id": 1,
                "images": Array [
                  "https://es31-server.appspot.com/six-cities/static/hotel/18.jpg",
                  "https://es31-server.appspot.com/six-cities/static/hotel/4.jpg",
                  "https://es31-server.appspot.com/six-cities/static/hotel/12.jpg",
                  "https://es31-server.appspot.com/six-cities/static/hotel/20.jpg",
                  "https://es31-server.appspot.com/six-cities/static/hotel/17.jpg",
                  "https://es31-server.appspot.com/six-cities/static/hotel/15.jpg",
                  "https://es31-server.appspot.com/six-cities/static/hotel/2.jpg",
                  "https://es31-server.appspot.com/six-cities/static/hotel/8.jpg",
                  "https://es31-server.appspot.com/six-cities/static/hotel/5.jpg",
                  "https://es31-server.appspot.com/six-cities/static/hotel/6.jpg",
                  "https://es31-server.appspot.com/six-cities/static/hotel/1.jpg",
                  "https://es31-server.appspot.com/six-cities/static/hotel/7.jpg",
                  "https://es31-server.appspot.com/six-cities/static/hotel/19.jpg",
                  "https://es31-server.appspot.com/six-cities/static/hotel/14.jpg",
                ],
                "isFavorite": false,
                "isPremium": true,
                "location": Object {
                  "latitude": 51.214402,
                  "longitude": 6.764314000000001,
                  "zoom": 16,
                },
                "maxAdults": 8,
                "previewImage": "https://es31-server.appspot.com/six-cities/static/hotel/1.jpg",
                "price": 393,
                "rating": 2.7,
                "title": "Waterfront with extraordinary view",
                "type": "hotel",
              }
            }
          />
          <ReviewsList
            reviews={
              Array [
                Object {
                  "comment": "A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.",
                  "date": "2019-05-08T14:13:56.569Z",
                  "id": 1,
                  "rating": 4,
                  "user": Object {
                    "avatarUrl": "img/1.png",
                    "id": 4,
                    "isPro": false,
                    "name": "Max",
                  },
                },
              ]
            }
          />
        </div>
      </div>
      <section
        className="property__map map"
      >
        <MapCity
          activeOffer={
            Object {
              "bedrooms": 2,
              "city": Object {
                "location": Object {
                  "latitude": 51.225402,
                  "longitude": 6.776314,
                  "zoom": 13,
                },
                "name": "Amsterdam",
              },
              "description": "I am happy to welcome you to my apartment in the city center! Three words: location, cosy and chic!",
              "goods": Array [
                "Breakfast",
                "Washer",
                "Laptop friendly workspace",
              ],
              "host": Object {
                "avatarUrl": "img/avatar-angelina.jpg",
                "id": 25,
                "isPro": true,
                "name": "Angelina",
              },
              "id": 1,
              "images": Array [
                "https://es31-server.appspot.com/six-cities/static/hotel/18.jpg",
                "https://es31-server.appspot.com/six-cities/static/hotel/4.jpg",
                "https://es31-server.appspot.com/six-cities/static/hotel/12.jpg",
                "https://es31-server.appspot.com/six-cities/static/hotel/20.jpg",
                "https://es31-server.appspot.com/six-cities/static/hotel/17.jpg",
                "https://es31-server.appspot.com/six-cities/static/hotel/15.jpg",
                "https://es31-server.appspot.com/six-cities/static/hotel/2.jpg",
                "https://es31-server.appspot.com/six-cities/static/hotel/8.jpg",
                "https://es31-server.appspot.com/six-cities/static/hotel/5.jpg",
                "https://es31-server.appspot.com/six-cities/static/hotel/6.jpg",
                "https://es31-server.appspot.com/six-cities/static/hotel/1.jpg",
                "https://es31-server.appspot.com/six-cities/static/hotel/7.jpg",
                "https://es31-server.appspot.com/six-cities/static/hotel/19.jpg",
                "https://es31-server.appspot.com/six-cities/static/hotel/14.jpg",
              ],
              "isFavorite": false,
              "isPremium": true,
              "location": Object {
                "latitude": 51.214402,
                "longitude": 6.764314000000001,
                "zoom": 16,
              },
              "maxAdults": 8,
              "previewImage": "https://es31-server.appspot.com/six-cities/static/hotel/1.jpg",
              "price": 393,
              "rating": 2.7,
              "title": "Waterfront with extraordinary view",
              "type": "hotel",
            }
          }
          changeZoom={true}
          city={
            Object {
              "location": Object {
                "latitude": 51.225402,
                "longitude": 6.776314,
                "zoom": 13,
              },
              "name": "Amsterdam",
            }
          }
          offers={
            Array [
              Object {
                "bedrooms": 2,
                "city": Object {
                  "location": Object {
                    "latitude": 51.225402,
                    "longitude": 6.776314,
                    "zoom": 13,
                  },
                  "name": "Amsterdam",
                },
                "description": "I am happy to welcome you to my apartment in the city center! Three words: location, cosy and chic!",
                "goods": Array [
                  "Breakfast",
                  "Washer",
                  "Laptop friendly workspace",
                ],
                "host": Object {
                  "avatarUrl": "img/avatar-angelina.jpg",
                  "id": 25,
                  "isPro": true,
                  "name": "Angelina",
                },
                "id": 1,
                "images": Array [
                  "https://es31-server.appspot.com/six-cities/static/hotel/18.jpg",
                  "https://es31-server.appspot.com/six-cities/static/hotel/4.jpg",
                  "https://es31-server.appspot.com/six-cities/static/hotel/12.jpg",
                  "https://es31-server.appspot.com/six-cities/static/hotel/20.jpg",
                  "https://es31-server.appspot.com/six-cities/static/hotel/17.jpg",
                  "https://es31-server.appspot.com/six-cities/static/hotel/15.jpg",
                  "https://es31-server.appspot.com/six-cities/static/hotel/2.jpg",
                  "https://es31-server.appspot.com/six-cities/static/hotel/8.jpg",
                  "https://es31-server.appspot.com/six-cities/static/hotel/5.jpg",
                  "https://es31-server.appspot.com/six-cities/static/hotel/6.jpg",
                  "https://es31-server.appspot.com/six-cities/static/hotel/1.jpg",
                  "https://es31-server.appspot.com/six-cities/static/hotel/7.jpg",
                  "https://es31-server.appspot.com/six-cities/static/hotel/19.jpg",
                  "https://es31-server.appspot.com/six-cities/static/hotel/14.jpg",
                ],
                "isFavorite": false,
                "isPremium": true,
                "location": Object {
                  "latitude": 51.214402,
                  "longitude": 6.764314000000001,
                  "zoom": 16,
                },
                "maxAdults": 8,
                "previewImage": "https://es31-server.appspot.com/six-cities/static/hotel/1.jpg",
                "price": 393,
                "rating": 2.7,
                "title": "Waterfront with extraordinary view",
                "type": "hotel",
              },
              Object {
                "bedrooms": 2,
                "city": Object {
                  "location": Object {
                    "latitude": 51.225402,
                    "longitude": 6.776314,
                    "zoom": 13,
                  },
                  "name": "Amsterdam",
                },
                "description": "I am happy to welcome you to my apartment in the city center! Three words: location, cosy and chic!",
                "goods": Array [
                  "Breakfast",
                  "Washer",
                  "Laptop friendly workspace",
                ],
                "host": Object {
                  "avatarUrl": "img/avatar-angelina.jpg",
                  "id": 25,
                  "isPro": true,
                  "name": "Angelina",
                },
                "id": 3,
                "images": Array [
                  "https://es31-server.appspot.com/six-cities/static/hotel/18.jpg",
                  "https://es31-server.appspot.com/six-cities/static/hotel/4.jpg",
                  "https://es31-server.appspot.com/six-cities/static/hotel/12.jpg",
                  "https://es31-server.appspot.com/six-cities/static/hotel/20.jpg",
                  "https://es31-server.appspot.com/six-cities/static/hotel/17.jpg",
                  "https://es31-server.appspot.com/six-cities/static/hotel/15.jpg",
                  "https://es31-server.appspot.com/six-cities/static/hotel/2.jpg",
                  "https://es31-server.appspot.com/six-cities/static/hotel/8.jpg",
                  "https://es31-server.appspot.com/six-cities/static/hotel/5.jpg",
                  "https://es31-server.appspot.com/six-cities/static/hotel/6.jpg",
                  "https://es31-server.appspot.com/six-cities/static/hotel/1.jpg",
                  "https://es31-server.appspot.com/six-cities/static/hotel/7.jpg",
                  "https://es31-server.appspot.com/six-cities/static/hotel/19.jpg",
                  "https://es31-server.appspot.com/six-cities/static/hotel/14.jpg",
                ],
                "isFavorite": false,
                "isPremium": true,
                "location": Object {
                  "latitude": 51.214402,
                  "longitude": 6.764314000000001,
                  "zoom": 16,
                },
                "maxAdults": 8,
                "previewImage": "https://es31-server.appspot.com/six-cities/static/hotel/1.jpg",
                "price": 393,
                "rating": 2.7,
                "title": "Waterfront with extraordinary view",
                "type": "hotel",
              },
            ]
          }
        />
      </section>
    </section>
    <div
      className="container"
    >
      <section
        className="near-places places"
      >
        <h2
          className="near-places__title"
        >
          Other places in the neighbourhood
        </h2>
        <div
          className="near-places__list places__list"
        >
          <OfferList
            needLink={true}
            offers={
              Array [
                Object {
                  "bedrooms": 2,
                  "city": Object {
                    "location": Object {
                      "latitude": 51.225402,
                      "longitude": 6.776314,
                      "zoom": 13,
                    },
                    "name": "Amsterdam",
                  },
                  "description": "I am happy to welcome you to my apartment in the city center! Three words: location, cosy and chic!",
                  "goods": Array [
                    "Breakfast",
                    "Washer",
                    "Laptop friendly workspace",
                  ],
                  "host": Object {
                    "avatarUrl": "img/avatar-angelina.jpg",
                    "id": 25,
                    "isPro": true,
                    "name": "Angelina",
                  },
                  "id": 1,
                  "images": Array [
                    "https://es31-server.appspot.com/six-cities/static/hotel/18.jpg",
                    "https://es31-server.appspot.com/six-cities/static/hotel/4.jpg",
                    "https://es31-server.appspot.com/six-cities/static/hotel/12.jpg",
                    "https://es31-server.appspot.com/six-cities/static/hotel/20.jpg",
                    "https://es31-server.appspot.com/six-cities/static/hotel/17.jpg",
                    "https://es31-server.appspot.com/six-cities/static/hotel/15.jpg",
                    "https://es31-server.appspot.com/six-cities/static/hotel/2.jpg",
                    "https://es31-server.appspot.com/six-cities/static/hotel/8.jpg",
                    "https://es31-server.appspot.com/six-cities/static/hotel/5.jpg",
                    "https://es31-server.appspot.com/six-cities/static/hotel/6.jpg",
                    "https://es31-server.appspot.com/six-cities/static/hotel/1.jpg",
                    "https://es31-server.appspot.com/six-cities/static/hotel/7.jpg",
                    "https://es31-server.appspot.com/six-cities/static/hotel/19.jpg",
                    "https://es31-server.appspot.com/six-cities/static/hotel/14.jpg",
                  ],
                  "isFavorite": false,
                  "isPremium": true,
                  "location": Object {
                    "latitude": 51.214402,
                    "longitude": 6.764314000000001,
                    "zoom": 16,
                  },
                  "maxAdults": 8,
                  "previewImage": "https://es31-server.appspot.com/six-cities/static/hotel/1.jpg",
                  "price": 393,
                  "rating": 2.7,
                  "title": "Waterfront with extraordinary view",
                  "type": "hotel",
                },
                Object {
                  "bedrooms": 2,
                  "city": Object {
                    "location": Object {
                      "latitude": 51.225402,
                      "longitude": 6.776314,
                      "zoom": 13,
                    },
                    "name": "Amsterdam",
                  },
                  "description": "I am happy to welcome you to my apartment in the city center! Three words: location, cosy and chic!",
                  "goods": Array [
                    "Breakfast",
                    "Washer",
                    "Laptop friendly workspace",
                  ],
                  "host": Object {
                    "avatarUrl": "img/avatar-angelina.jpg",
                    "id": 25,
                    "isPro": true,
                    "name": "Angelina",
                  },
                  "id": 3,
                  "images": Array [
                    "https://es31-server.appspot.com/six-cities/static/hotel/18.jpg",
                    "https://es31-server.appspot.com/six-cities/static/hotel/4.jpg",
                    "https://es31-server.appspot.com/six-cities/static/hotel/12.jpg",
                    "https://es31-server.appspot.com/six-cities/static/hotel/20.jpg",
                    "https://es31-server.appspot.com/six-cities/static/hotel/17.jpg",
                    "https://es31-server.appspot.com/six-cities/static/hotel/15.jpg",
                    "https://es31-server.appspot.com/six-cities/static/hotel/2.jpg",
                    "https://es31-server.appspot.com/six-cities/static/hotel/8.jpg",
                    "https://es31-server.appspot.com/six-cities/static/hotel/5.jpg",
                    "https://es31-server.appspot.com/six-cities/static/hotel/6.jpg",
                    "https://es31-server.appspot.com/six-cities/static/hotel/1.jpg",
                    "https://es31-server.appspot.com/six-cities/static/hotel/7.jpg",
                    "https://es31-server.appspot.com/six-cities/static/hotel/19.jpg",
                    "https://es31-server.appspot.com/six-cities/static/hotel/14.jpg",
                  ],
                  "isFavorite": false,
                  "isPremium": true,
                  "location": Object {
                    "latitude": 51.214402,
                    "longitude": 6.764314000000001,
                    "zoom": 16,
                  },
                  "maxAdults": 8,
                  "previewImage": "https://es31-server.appspot.com/six-cities/static/hotel/1.jpg",
                  "price": 393,
                  "rating": 2.7,
                  "title": "Waterfront with extraordinary view",
                  "type": "hotel",
                },
              ]
            }
          />
        </div>
      </section>
    </div>
  </main>
</div>
`;
